### DISCUSSION TITLE
Custom server route returns 404

### DISCUSSION THREAD

**Discussion by sospedra** [vercel/next.js#13692] [Help] [answered]
# Bug report

## Describe the bug

1. When using a custom server
2. The new paths on that server work on development
3. But when deployed with Vercel they return a 404

## To Reproduce

1. Check this https://peecto.now.sh/ping (it's a 404)
2. This [is the code](https://github.com/sospedra/peecto/commit/7f97fb9e609b660c09fca13a066f6a7cb6d03a90)
3. If you pull and run the project it'll work

## Expected behavior

That works the same in prod as in devel

## Aditional

I tried multiple approaches: catch-all with `*`, `next-routes`, etc. But nothing seems to work.
And, honestly, it's kinda painful that fails after the deployment because testing it that way is harsh 😞 

ty

**Comment by jamesmosier** [comment, marked as answer]
The last I heard (via [this comment](https://github.com/vercel/next.js/issues/9397#issuecomment-556215227)) is that Vercel will not work properly with a custom server.

**Reply by sospedra** [reply]
Oh! Wow, I spent a long time digging for this kind of information. I guess this should be on the docs.

Anyway, does anyone know how Vercel recommends tackling a problem like mine? Where you need a different server approach (in this case WebSocket and WebRTC signalling)?

A different deploy? 🤔

**Reply by sospedra** [reply]
Copy and paste the quote for future searches purposes:

> Custom servers do not work on ZEIT Now. You'll need to use our built-in router and potentially the upcoming Custom Routes RFC: #9081.
> 
> We'll be removing most of the custom server examples soon-ish. They shouldn't be needed for most applications!

**Comment by sospedra** [comment]
I simply **moved the handshake to a Heroku dyno**. And I'll close the issue since the example is not accurate anymore (deployment and code changed). But I'd love to see some changes in the docs regarding these situations. Let me know if I can do something to help.

cc @timneutkens @Timer
